tests/debugger/test_debugger_exception_replay.py

```diff
@@ -3,26 +3,55 @@
 # Copyright 2021 Datadog, Inc.
 
 import tests.debugger.utils as base
-import time, json, os
+import json, os
 from utils import scenarios, interfaces, weblog, features
+from utils.tools import logger
 
 
 @features.debugger_exception_replay
 @scenarios.debugger_exception_replay
 class Test_Debugger_Exception_Replay(base._Base_Debugger_Test):
     tracer = None
+    snapshot = None
+    method = None
 
-    def _setup(self, request_path):
-        self.weblog_responses = [weblog.get(request_path)]
-        time.sleep(1)
-        self.weblog_responses.append(weblog.get(request_path))
+    def _setup(self, request_path, method):
+        self.snapshot = None
+        self.weblog_responses = set()
+        self.method = method
+
+        retries = 0
+        max_retries = 60
+
+        while not self.snapshot and retries < max_retries:
+            logger.debug(f"Waiting for snapshot, retry #{retries}")
+
+            self.weblog_responses.add(weblog.get(request_path))
+            interfaces.agent.wait_for(self._wait_for_snapshot_received, timeout=1)
+
+            retries += 1
+
+    def _wait_for_snapshot_received(self, data):
+        if data["path"] == base._LOGS_PATH:
+
+            contents = data["request"].get("content", []) or []
+            if contents:
+                for content in contents:
+                    snapshot = content.get("debugger", {}).get("snapshot") or content.get("debugger.snapshot")
+                    if snapshot and snapshot["probe"]["location"]["method"].lower().replace("_", "") == self.method:
+                        self.snapshot = snapshot
+
+                        logger.debug("Snapshot received")
+                        return True
+
+        return False
```

Comment:
```suggestion
        def snapshot_received(data):
            if data["path"] == base._LOGS_PATH:

                contents = data["request"].get("content", []) or []
                if contents:
                    for content in contents:
                        snapshot = content.get("debugger", {}).get("snapshot") or content.get("debugger.snapshot")
                        if snapshot and snapshot["probe"]["location"]["method"].lower().replace("_", "") == self.method:
                            self.snapshot = snapshot

                            logger.debug("Snapshot received")
                            return True

            return False


        logger.debug(f"Waiting for snapshot, retry #{retries}")

        self.weblog_responses.add(weblog.get(request_path))
        interfaces.agent.wait_for(snapshot_received, timeout=60)
```

Reply:
It will work great for python, but not for dotnet and java, since we have to send at least two requests to make exception replay work.